hw/drivers/sensors/bma253/include/bma253/bma253.h

```diff
@@ -197,8 +201,29 @@ struct bma253_private_driver_data {
     uint8_t int_num;
     uint8_t int_route;
     uint8_t int_ref_cnt;
+
+
+    //zg: check for potential improvement
+    uint8_t fifo_buf[31 * 6];
```

Comment:
There is no need for the driver to define the FIFO buffer. The app can define it and the driver can just have a ptr to it and fill it up. `hw/drivers/sensors/lis2dh12/src/lis2dh12.c` has a `lis2dh12_stream_read` function which you can use as reference. It's quite necessary for it to remain that way or we might have issues with completing abstracting the application from the driver.

Reply:
Hello @vrahane 

Thanks for the info, actually, this buf is not for filling up users's data (structured data), this is the buffer to hold the raw data (binary and non-structured) from the FIFO and to take care of the worst sense scenario (when fifo is full). 

Otherwise, all other reviews are agreed upon and fixed/changed correspondingly.